### PR TITLE
feat: extract images in markdown contents when using vision models

### PR DESCRIPTION
**Pull request by DynamesC**
# Pull Request

## PR Description

Extract images in markdown contents and feed to models in formatted style when using vision models

## Linked Issue

No linked issue

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

## Additional Information

Provide any other important information about this PR here.

### CODE REVIEW

**Review by taskingaijc — DISMISSED** [review]
LGTM

**Review by taskingaijc — DISMISSED** [review]
LGTM

**Review by jameszyao — DISMISSED** [review]
LGTM

File: inference/providers/openai/chat_completion.py [hunk 1]
```diff
@@ -48,16 +84,17 @@ def _build_openai_message(message: ChatCompletionMessage):
...
     stream: bool,
     provider_model_id: str,
     configs: ChatCompletionModelConfiguration,
     function_call: Optional[str],
     functions: Optional[List[ChatCompletionFunction]],
+    vision_support: bool = False,
 ):
     # Convert ChatCompletionMessages to the required format
-    formatted_messages = [_build_openai_message(msg) for msg in messages]
+    formatted_messages = [await _build_openai_message(msg, vision_support) for msg in messages]
```
Comment (jameszyao) [inline inference/providers/openai/chat_completion.py:97]: Only convert the most recent message that includes inline images

**Review by jameszyao — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**